app/code/Magento/Inventory/Model/SourceTypeLinkManagement.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryApi\Model\SourceTypeLinkManagementInterface;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\Inventory\Model\ResourceModel\SourceTypeLink;
+use Magento\Inventory\Model\ResourceModel\SourceTypeLink\CollectionFactory;
+
+/**
+ * @inheritdoc
+ */
+class SourceTypeLinkManagement implements SourceTypeLinkManagementInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var CollectionProcessorInterface
+     */
+    private $collectionProcessor;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $typeLinkCollectionFactory;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param CollectionProcessorInterface $collectionProcessor
+     * @param CollectionFactory $carrierLinkCollectionFactory
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        CollectionProcessorInterface $collectionProcessor,
+        CollectionFactory $typeLinkCollectionFactory,
+        SearchCriteriaBuilder $searchCriteriaBuilder
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->collectionProcessor = $collectionProcessor;
+        $this->typeLinkCollectionFactory = $typeLinkCollectionFactory;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function saveTypeLinksBySource(SourceInterface $source, $type_code): void
+    {
+        $this->deleteCurrentTypeLinks($source);
+
+        $this->saveNewTypeLinks($source, $type_code);
+    }
+
+    /**
+     * @param SourceInterface $source
+     * @return void
+     */
+    private function deleteCurrentTypeLinks(SourceInterface $source)
+    {
+        $connection = $this->resourceConnection->getConnection();
```

Comment:
It's best to operate with data storage connection from the Data Access layer, model shouldn't contain this kind of logic

Reply:
i was tried use app/code/Magento/Inventory/Model/SourceCarrierLinkManagement.php as exemple.